generators/client/templates/angular/src/main/webapp/app/shared/alert/alert-error.component.ts.ejs

```diff
@@ -118,33 +138,32 @@ export class <%=jhiPrefixCapitalized%>AlertErrorComponent implements OnDestroy {
 
     addErrorAlert(message, key?, data?) {
         <%_ if (enableTranslation) { _%>
-        key = (key && key !== null) ? key : message;
-        this.alerts.push(
-            this.alertService.addAlert(
-                {
-                    type: 'danger',
-                    msg: key,
-                    params: data,
-                    timeout: 5000,
-                    toast: this.alertService.isToast(),
-                    scoped: true
-                },
-                this.alerts
-            )
-        );
+        const info = (key && key !== null) ? key : message;
+        <%_ } else { _%>
+        const info = message;
+        <%_ } _%>
+
+        const newAlert: JhiAlert = {
+            type: 'danger',
+            msg: info,
+            <%_ if (enableTranslation) { _%>
+            params: data,
+            <%_ } _%>
+            timeout: this.timeout,
+            toast: this.alertService.isToast(),
+            scoped: true
+        };
+
+        const alertsMessage: string[] = this.alerts.map(alert => alert.msg);
+        <%_ if (enableTranslation) { _%>
+        const newAlertMessage = this.translateService.instant(newAlert.msg);
         <%_ } else { _%>
-        this.alerts.push(
-            this.alertService.addAlert(
-                {
-                    type: 'danger',
-                    msg: message,
-                    timeout: 5000,
-                    toast: this.alertService.isToast(),
-                    scoped: true
-                },
-                this.alerts
-            )
-        );
+        const newAlertMessage = newAlert.msg;
         <%_ } _%>
+
+        if (!alertsMessage.includes(newAlertMessage) || newAlert.timeout !== 0) {
+            // this.alerts.push(this.alertService.addAlert(newAlert, this.alerts));
```

Comment:
we cant have these in generated code, please cleanup

Reply:
Sure! My PR is still WIP and my changes will need to update the ng-jhipster project. That was just to check if tests passed. I'm waiting for some feedbacks to know if I continue or not